src/test/groovy/org/gradle/wrapperupgrade/GradleWrapperUpgradePluginFuncTest.groovy

```diff
@@ -125,6 +125,51 @@ wrapperUpgrade {
         result.output.contains('Reusing configuration cache.')
     }
 
+    def "upgrade wrapper on wrapper-upgrade-gradle-plugin with dry run and optional Git arguments"() {
+        given:
+        buildFile.text = """
+
+plugins {
+    id 'base'
+    id 'org.gradle.wrapper-upgrade'
+}
+
+wrapperUpgrade {
+    gradle {
+        'wrapper-upgrade-gradle-plugin-for-func-tests' {
+            repo = 'gradle/wrapper-upgrade-gradle-plugin'
+            baseBranch = 'func-test-do-not-delete'
+            dir = 'samples/gradle'
+            options {
+                gitCommitExtraArgs = ['--signoff', '--date="Wed Mar 23 15:00:00 CET 2022"']
+            }
+        }
+    }
+}
+        """
+        when:
+        def result = GradleRunner.create()
+            .withProjectDir(testProjectDir)
+            .withPluginClasspath()
+            .withGradleVersion(determineGradleVersion().version)
+            .withArguments('clean', 'upgradeGradleWrapperAll', '-DwrapperUpgrade.dryRun', '-DwrapperUpgrade.unsignedCommits')
```

Comment:
I would remove the `-DwrapperUpgrade.unsignedCommits` flag here (see comment above). passing in the `--date` option will suffice to very that the options are working well.